docs/source/deployment/index.rst

```diff
@@ -169,3 +169,144 @@ Almost all functionalities available in MLflow deployment can also be accessed v
+The image will contain your model and dependencies and has an entrypoint to start the inference server. Similarly to the `serve` API,
+you can send a test request and validate the response.
+Please refer to `CLI reference <../cli.html#mlflow-models>`_ for more detailed usage for the `build-docker` API.
+
+.. code-block:: bash
+
+    mlflow models build-docker -m runs:/<run_id>/model -n <image_name>
+
+    docker run -p <port>:8080 <image_name>
+
+    # In another terminal
+    curl -X POST -H "Content-Type: application/json" \
+        --data '{"inputs": [[1, 2], [3, 4]]}' \
+        http://localhost:<port>/invocations
+
+
+How to fix dependency errors when serving my model?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+One common issue during model deployment is dependency issue. When logging/saving your model, MLflow tries to infer the
+model dependencies and save them as part of the MLflow Model metadata. However, this inference is not always accurate and may
+miss some dependencies. This can cause errors when serving your model, such as "ModuleNotFoundError" or "ImportError". Here 
+is the steps to fix missing dependencies error.
+
+1. Check the missing dependencies
+*********************************
+The missing dependencies are listed in the error message. For example, if you see the following error message:
+
+.. code-block:: bash
+
+    ModuleNotFoundError: No module named 'cv2'
+
+2. Try adding the dependencies using the `predict` API
+******************************************************
+Now that you know the missing dependencies, you can create a new model version with the correct dependencies.
+However, creating a new model for trying new dependencies might be a bit tedious, particularly because you may need to
+iterate multiple times to find the correct solution. Instead, you can use the `predict` API to test your change without
+actually mutating the model.
+
+To do so, use the `pip-requirements-override` option to specify pip dependencies like `opencv-python==4.8.0`.
+
+.. tabs::
+
+    .. code-tab:: bash
+
+        mlflow models predict \
+            -m runs:/<run_id>/model \
+            -i <input_path> \
+            --pip-requirements-override opencv-python==4.8.0
+
+    .. code-tab:: python
+
+        import mlflow
+
+        mlflow.model.predict(
+            model_uri="runs:/<run_id>/model",
+            input_data=<input_data>,
+            pip_requirements="opencv-python==4.8.0",
+        )
+
+The specified dependencies will be installed to the virtual environment in addition to (or instead of) the dependencies
+defined in the model metadata. Since this doesn't mutate the model, you can iterate quickly and safely to find the correct dependencies.
+
+3. Update the model metadata
+****************************
+Once you find the correct dependencies, you can create a new model with the correct dependencies.
+To do so, specify either `pip_requirements` or `extra_pip_requirements` option when logging the model.
```

Comment:
Specifying `pip_requirements` will get rid of the other requirements that MLflow infers, right? I think users always want `extra_pip_requirements` to solve missing dependency issues.